models/mock.go

```diff
@@ -4,57 +4,85 @@ type MockDB struct {
 	Conversation *Conversation
 	Mapping      *UserConversationMapping
 	Mappings     []*UserConversationMapping
-	Error        error
+	Errors       []error
 
+	Count           int
 	ConversationIDs []int64
 	UserIDs         []int64
 }
 
+func NewMockDB(
+	conversation *Conversation,
+	mapping *UserConversationMapping,
+	mappings []*UserConversationMapping,
+	errors []error,
+) *MockDB {
+	return &MockDB{
+		Conversation:    conversation,
+		Mapping:         mapping,
+		Mappings:        mappings,
+		Errors:          errors,
+		Count:           0,
+		ConversationIDs: make([]int64, 0),
+		UserIDs:         make([]int64, 0),
+	}
+}
+
+func (db *MockDB) getError() error {
+	if db.Errors == nil {
+		return nil
+	} else {
+		err := db.Errors[db.Count]
```

Comment:
Should probably check that `db.Count` doesn't exceed the size of `db.Errors`

Reply:
I disagree